src/main.rs

```diff
@@ -1,124 +1,93 @@
-use std::env;
-use std::path::{Path, PathBuf};
+use std::path::PathBuf;
 use std::time::Instant;
 
+use cli::{Cli, Command};
 use utils::net::{get_available_port, port_is_available};
 
+use clap::Parser;
+
 mod cli;
 mod cmd;
 mod console;
 mod prompt;
 
 fn main() {
-    let matches = cli::build_cli().get_matches();
+    let cli = Cli::parse();
+    let cli_dir: PathBuf = cli.root.canonicalize().unwrap_or_else(|_| {
+        panic!("Could not find canonical path of root dir: {}", cli.root.display())
+    });
 
-    let root_dir = match matches.value_of("root").unwrap() {
-        "." => env::current_dir().unwrap(),
-        path => PathBuf::from(path)
-            .canonicalize()
-            .unwrap_or_else(|_| panic!("Cannot find root directory: {}", path)),
-    };
-    let config_file = match matches.value_of("config") {
-        Some(path) => PathBuf::from(path)
-            .canonicalize()
-            .unwrap_or_else(|_| panic!("Cannot find config file: {}", path)),
-        None => root_dir.join("config.toml"),
-    };
+    let root_dir = cli_dir
+        .ancestors()
+        .find_map(|a| if a.join(&cli.config).exists() { Some(a) } else { None })
+        .unwrap_or_else(|| panic!("could not find directory containing config file"));
```

Comment:
@Keats I think something in here breaks `zola init`.  I tried init today on the next branch and it immediately panics with this error.

Reply:
Hm, I'll fix it but we should really have some test for the cli :(